ui/admin/app/routes/scopes/scope/session-recordings/index.js

```diff
@@ -4,5 +4,160 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
 
-export default class ScopesScopeSessionRecordingsIndexRoute extends Route {}
+export default class ScopesScopeSessionRecordingsIndexRoute extends Route {
+  // =services
+  @service store;
+  @service router;
+  @service can;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    times: {
+      refreshModel: true,
+      replace: true,
+    },
+    users: {
+      refreshModel: true,
+      replace: true,
+    },
+    scopes: {
+      refreshModel: true,
+      replace: true,
+    },
+    targets: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  allSessionRecordings;
+
+  /**
+   * Load all session recordings.
+   * @return {Promise<{ totalItems: number, sessionRecordings: [SessionRecordingModel], sessionRecordingsExist: boolean, storageBucketsExist: boolean }>}
+   */
+  async model({ search, times, users, scopes, targets, page, pageSize }) {
+    const scope = this.modelFor('scopes.scope');
+    const { id: scope_id } = scope;
+    let sessionRecordings;
+    let totalItems = 0;
+    let sessionRecordingsExist = false;
+    let storageBucketsExist = false;
+    const filters = {
+      scope_id: [{ equals: scope_id }],
+      created_time: [],
+      'create_time_values.user.id': [],
+      'create_time_values.target.scope.id': [],
+      'create_time_values.target.id': [],
+    };
+    times.forEach((time) => {
+      filters.created_time.push({ gte: new Date(time) });
+    });
+    users.forEach((user) => {
+      filters['create_time_values.user.id'].push({ equals: user });
+    });
+    scopes.forEach((scope) => {
+      filters['create_time_values.target.scope.id'].push({ equals: scope });
+    });
+    targets.forEach((target) => {
+      filters['create_time_values.target.id'].push({ equals: target });
+    });
+
+    if (
+      this.can.can('list scope', scope, {
+        collection: 'session-recordings',
+      })
+    ) {
+      const queryOptions = {
+        scope_id,
+        query: { search, filters },
+        page,
+        pageSize,
+      };
+
+      sessionRecordings = await this.store.query(
+        'session-recording',
+        queryOptions,
+      );
+      totalItems = sessionRecordings.meta?.totalItems;
+      // Query all session reocordings for filtering values if entering route for the first time
+      if (!this.allSessionRecordings) {
+        await this.getAllSessionRecordings(scope_id);
+      }
+      sessionRecordingsExist = !!this.allSessionRecordings.length;
```

Comment:
I don't think we've traditionally used `!!`, haven't we mainly used `Boolean`?

Reply:
Yep we have. Updated this as well.